TERMS.md

```diff
@@ -0,0 +1,97 @@
+	- Namespaces are used to ensure that local names will not clash with identically named items in other namespace.
+	- Namespaces usually also separate what objects can be *seen* or *reached* from within a part of a program.
+	- Namespaces can contain other Namespaces, creating a hierarchy.
+	- How visibility/reachability is handled *might* be specific to the target language or environment, but VSC generally assumes that items within different (non-parent) namespaces are isolated from each other unless otherwise specified, and that common hierarchy rules apply (i.e. everything within a child namespace can see and reach items in any of its parent namespaces without a specific required statement of reference or inclusion).
+- **Interface**
+	- (General)
+		- very broad term...
+	- (VSC)
+	- A VSC **Interface** is *primarily* defined by a collection of **Methods**.
+	- An Interface name is not itself a Namespace but must be defined within a Namespace.
+	- All Methods defined or referred to by an Interface are required to be within the same Namespace, i.e. it is not possible to build an Interface by referring to Methods that have been defined in different Namespaces.
+	- The VSC interface definition language strives to usable in all areas of a computing system.   The VSC Interface object is therefore a pure software function focused definition, which does not define details around communication hardware, target language, communication protocols etc.  Such information is added through supplementary information later.
+- **Method**
+	- A method is a function with a (mandatory) name, and a set of **input parameters**, **output parameters**, and **error conditions**, all of which are optional and may be empty/undefined.
+	- Like most other objects, a Method is always defined in the context of a Namespace, but this is most often implicitly determined because the Method is defined within the context of an Interface/Service
+- **Error**
+	- A VSC Error definition is an error condition that can appear while (attempting to) execute a given Method.
+	- An error is defined by referencing one or more **Type**s that define the data that shall be communicated when an error occurs.  Enumeration types are common, but it could be any defined type.
+	- There can be more than one error type defined for a method
+	- It is not specified by VSC how errors are propagated - this translation is target language/environment specific.
+		- It may be added as return/output parameters in one programming language and translated to **Exception**-handling in another.  Communication protocols may implement their own error-condition channels and approaches.
+
+
+----
+
+NEW 2022-08-08:
+
+----
+
+## Properties
+
+(N.B. In Franca IDL these are called Attributes)
+
+- A **Property** is an observable data item.  It belongs in a **Namespace**, has a canonical name in addition to possible **aliases** (i.e. alternative names), and a data type.
```

Comment:
Belongs in a namespace or in an interface?

Reply:
Yes, I think that's right. Properties probably need to be inside an Interface (which indirectly defines that they are within the same Namespace as the Interface was)